### ISSUE TITLE
make_heap code object inequality at import time

### ISSUE THREAD

**Issue by JesseRMeyer** [lmcinnes/pynndescent#250]
In attempting to go through the introductory material for UMAP, I hit the following exception with `import umap`:

`<code object <module> at ..., file "/path/to/keras/lib/python3.12/path/to/pynndescent/sparse_nndescent.py", line 1> != <code object make_heap at ..., file "/path/to/keras/lib/python3.12/path/to/pynndescent/utils.py", line 171>`.

I'm at a loss on how to debug this.  Note that the venv here is called "keras".

WSL2
Python 3.12
scikit-learn 1.6
umap-learn 0.5.7
scipy 1.14.1
pynndescent 0.5.13
numpy 2.0.2
numba 0.60.0
keras 3.6.0

**Comment by JesseRMeyer** [comment]
Note that I've since tried running this from a new venv without keras, and the same problem manifests.

**Comment by lmcinnes** [comment]
I have never seen this before. I'll have to see if I can reproduce anything like it locally.

**Comment by JesseRMeyer** [comment]
A member of our group reports that a very similar setup with Python 3.10 works as expected (both on Native Windows and through WSL).

**Comment by erathorn** [comment]
I have the same issue on my side with the following specifications:

- Linux
- Python 3.12
- scikit-learn=1.6.0
- scipy=1.15.0
- umap-learn=0.5.7
- pynndescent=0.5.13
- numpy=2.0.2
- numba=0.60.0

I installed everything via `mamba install -c conda-forge umap-learn`.

Maybe this helps in tracking down the issue.

**Comment by ratthapon** [comment]
I have this problem when I run a file using debugger (VS Code). Is there any recommended debugger?

**Comment by flying-sheep** [comment]
Yeah, without being able to use a debugger, I can’t maintain anything …

This seems to be the asssertion that gets triggered: https://github.com/microsoft/debugpy/blob/39879bdeeae5908a8f3527c77ba252e0930afabc/src/debugpy/_vendored/pydevd/_pydevd_sys_monitoring/_pydevd_sys_monitoring_cython.pyx#L562

Workaround: https://github.com/microsoft/debugpy/issues/1719

> Current workaround is to downgrade either numba to 0.59.1 or python debugger to v2024.10.0

(I think using Python 3.11 also works)

Upstream issue: https://github.com/numba/numba/issues/9760
